app/jobs/descmetadata_download_job.rb

```diff
@@ -64,11 +63,11 @@ def process_druid(current_druid, log, zip_file)
   # Queries DOR for a given druid, attempting up to MAX_TRIES times in case of failure.
   # @param [String]   druid   The ID of the object to find
   # @param [File]     log     Log file to write to
-  # @return  The Dor::Item corresponding to the given druid, or nil if none was found.
+  # @return [Cocina::Models::DRO,Cocina::Models::Collection,Cocina::Models::AdminPolicy,nil] cocina model instance corresponding to the given druid, or nil if none was found.
   def query_dor(druid, log)
     attempts ||= MAX_TRIES
-    Dor.find druid
-  rescue RestClient::RequestTimeout
+    Dor::Services::Client.object(druid).find
+  rescue Dor::Services::Client::NotFoundResponse, Dor::Services::Client::UnexpectedResponse
```

Comment:
Leave previous retry logic in place but use DSC-relevant exceptions. I could be convinced to remove this entirely but didn't seem necessary for the requested refactoring.

Reply:
I agree. I don't think we need this, but fine.  However, I don't think that repeating on these exceptions is the equivalent behavior to the timeout.  The equivalent would be a Faraday::TimeoutError

Reply:
@jcoyne Good call. Made that change and pushed it. Thanks!